extensions/amp-form/0.1/amp-form.js

```diff
@@ -292,11 +295,19 @@ export class AmpForm {
       user().warn(TAG, 'Variable substitutions disabled for non-canonical ' +
           'origin submit action: %s', this.form_);
     }
+
+    for (const k in formObject) {
+      formData['formFields[' + k + ']'] = formObject[k][0];
```

Comment:
The problem with joining is that it won't guarantee that the string will be splittable later, because form fields can contain any values, including the separator. There doesn't seem to be a universal solution, so I agree that joining is better, because it passes all the values.